nvflare/app_opt/job_launcher/job_launcher_spec.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import abstractmethod
+
+from nvflare.apis.fl_context import FLContext
+
+
+class JobHandleSpec:
+    @abstractmethod
+    def terminate(self):
```

Comment:
Suggest to have a grace_time, as we discussed.

Reply:
timeout used in the K8sJobLauncher. Made it an implementation specific control